packages/nimble-components/src/select/index.ts

```diff
@@ -1307,6 +1316,26 @@ export class Select
         this.setPositioning();
         this.focusAndScrollOptionIntoView();
     }
+
+    private observeSelectedOptionTextContent(): void {
+        this.selectedOptionObserver?.disconnect();
+
+        if (this.selectedIndex === -1) {
+            return;
+        }
+
+        const selectedOption = this.firstSelectedOption;
+        if (selectedOption) {
+            this.selectedOptionObserver = new MutationObserver(() => {
+                this.updateDisplayValue();
+            });
```

Comment:
```suggestion
          if (!this.selectedOptionObserver) {
              this.selectedOptionObserver = new MutationObserver(() => {
                  this.updateDisplayValue();
              });
          }
```

Reply:
Or instead of constructing it lazily we could construct it once as a readonly prop. Guessing most select's will have an option marked selected so probably not avoiding much making it lazy